src/test/java/ay2122s1_cs2103t_w16_2/btbb/model/client/predicate/ClientPredicateCollectionTest.java

```diff
@@ -0,0 +1,82 @@
+package ay2122s1_cs2103t_w16_2.btbb.model.client.predicate;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import ay2122s1_cs2103t_w16_2.btbb.model.client.Client;
+import ay2122s1_cs2103t_w16_2.btbb.testutil.ClientBuilder;
+
+public class ClientPredicateCollectionTest {
+    public static final NameContainsKeywordsPredicate NAME_ALICE_BOB_PREDICATE =
+            new NameContainsKeywordsPredicate(List.of("Alice", "Bob"));
+    public static final PhoneContainsKeywordsPredicate PHONE_9427_3217_PREDICATE =
+            new PhoneContainsKeywordsPredicate(List.of("9427", "3217"));
+    public static final AddressContainsKeywordsPredicate ADDRESS_YISHUN_GEYLANG_PREDICATE =
+            new AddressContainsKeywordsPredicate(List.of("Yishun", "Geylang"));
+    public static final EmailContainsKeywordsPredicate EMAIL_ALICE_BOB_GMAIL_PREDICATE =
+            new EmailContainsKeywordsPredicate(List.of("alice@gmail.com", "bob@gmail.com"));
+
+    public static void addPredicates(ClientPredicateCollection clientPredicateCollection,
+                               List<Predicate<Client>> clientPredicates) {
+        for (Predicate<Client> clientPredicate : clientPredicates) {
+            clientPredicateCollection.addClientPredicate(clientPredicate);
+        }
+    }
+
+    @Test
+    public void equals() {
```

Comment:
Add `different types -> returns false` and `null -> returns false`. Refer to other test files